Scripts/Runtime/Jobs/Collections/CollectionAccessControl.cs

```diff
@@ -0,0 +1,94 @@
+using System.Collections.Generic;
+using Unity.Entities;
+using Unity.Jobs;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Jobs
+{
+    /// <summary>
+    /// Helper class for managing access control to collections across multiple jobs.
+    /// </summary>
+    /// <remarks>
+    /// Unity will handle this for you if you're dealing with <see cref="EntityQuery"/> or <see cref="IComponentData"/>
+    /// but if it's just a native collection, you will have to manage your read/write access.
+    ///
+    /// The <see cref="IJob"/> safety system will throw errors if you handle it incorrectly but that's not a very
+    /// efficient way to develop and correct.
+    /// </remarks>
+    /// <typeparam name="TContext">A type to give context to allow for multiple keys of the same time (int, string etc)
+    /// to be used in a project.</typeparam>
+    /// <typeparam name="TKey">The type of key to use to get an instance of an <see cref="CollectionAccessController{TContext, TKey}"/></typeparam>
+    public static class CollectionAccessControl<TContext, TKey>
```

Comment:
Why did you choose to use this generics pattern? Would we better served by a single generic that accepts an Enum with an ID for the controller?

It's a bit of a weird pattern that generates a lot of copies of this class and the `CollectionAccessController` instances. 

Without generics we could do a mass dispose/remove call from this static class rather than having to call dispose on every generic variant.

Reply:
I had chosen it so that we were forcing the user to be a bit safer but your comments made me refactor things quite a bit so that it's cleaner for the end user to use. 

As an added bonus, all `CollectionAccessController`s can now be mass disposed from one API call regardless of context or key type.